src/hooks/category.ts

```diff
@@ -0,0 +1,54 @@
+import { useQuery } from 'react-query';
+import { List } from 'immutable';
+import { QueryClientConfig, UUID } from '../types';
+import * as Api from '../api';
+import { buildItemCategoryKey, buildItemsByCategoryKey, buildCategoriesKey, CATEGORY_INFO, CATEGORY_TYPES_KEY } from '../config/keys';
+
+export default (queryConfig: QueryClientConfig) => {
+  const { retry, cacheTime, staleTime } = queryConfig;
+  const defaultOptions = {
+    retry,
+    cacheTime,
+    staleTime,
+  };
+  
+  // get category types
+  const useCategoryTypes = () =>
+    useQuery({
+      queryKey: CATEGORY_TYPES_KEY,
+      queryFn: () => Api.getCategoryTypes(queryConfig).then((data) => List(data)),
+      ...defaultOptions,
+    });
+
+  // get categories
+  const useCategories = (typeId?: UUID[]) =>
+    useQuery({
+      queryKey: buildCategoriesKey(typeId),
+      queryFn: () => Api.getCategories(queryConfig, typeId).then((data) => List(data)),
+      ...defaultOptions,
+    });
+
+  const useCategoryInfo = (categoryId: UUID) =>
+    useQuery({
+      queryKey: CATEGORY_INFO,
+      queryFn: () => Api.getCategoryInfo(categoryId ,queryConfig).then((data) => (data)),
```

Comment:
You can use `Map` here -> you will probably have to update graasp-compose.

Reply:
Yes currently I'm getting all categories and use Map (in frontend), but I would like to have the endpoint here, just in case we will need it in the future!